src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemDatasetValidator.cs

```diff
@@ -96,9 +96,9 @@ protected static void ValidateProcedureStepState(DicomDataset dicomDataset, stri
                         FailureReasonCodes.ValidationFailure,
                         string.Format(
                             CultureInfo.InvariantCulture,
-                            DicomCoreResource.InvalidProcedureStepState,
-                            currentState,
+                            DicomCoreResource.InvalidProcedureStepStateTransition,
                             workitemInstanceUid,
+                            currentState,
```

Comment:
this method should take the current and requested states as parameters. Currently this only is for the Add scenario